analyzer/src/main/kotlin/managers/utils/NuGetSupport.kt

```diff
@@ -126,8 +126,25 @@ class NuGetSupport {
     private fun getAllPackageData(id: Identifier): NuGetAllPackageData {
         // Note: The package name in the URL is case-sensitive and must be lower-case!
         val lowerId = id.name.toLowerCase()
-        val dataUrl = "$registrationsBaseUrl/$lowerId/${id.version}.json"
-        val data = runBlocking { mapFromUrl<PackageData>(JSON_MAPPER, dataUrl) }
+
+        var data: PackageData? = null
+        for (baseUrl in registrationsBaseUrls) {
```

Comment:
We generally prefer the `registrationsBaseUrls.forEach { baseUrl ->` syntax.

Reply:
I used `for (baseUrl in registrationsBaseUrls) {` because I wanted to use `break` within the loop. That can be replaced with `return@forEach`.
But more importantly using `.forEach` prevents smart casting of `data` later below:
> Smart cast to 'NuGetAllPackageData.PackageData' is impossible, because 'data' is a local variable that is captured by a changing closure